tests/nest_tests/stdp_synapse_test.py

```diff
@@ -50,26 +54,35 @@ class NestSTDPSynapseTest(unittest.TestCase):
     def setUp(self):
         """Generate the model code"""
 
+        jit_codegen_opts = {"neuron_synapse_pairs": [{"neuron": "iaf_psc_exp",
+                                                      "synapse": "stdp",
+                                                      "post_ports": ["post_spikes"]}]}
+        if not nest_version.startswith("v2"):
+            jit_codegen_opts["neuron_parent_class"] = "StructuralPlasticityNode"
+            jit_codegen_opts["neuron_parent_class_include"] = "structural_plasticity_node.h"
+
         # generate the "jit" model (co-generated neuron and synapse), that does not rely on ArchivingNode
         generate_nest_target(input_path=["models/neurons/iaf_psc_exp.nestml", "models/synapses/stdp_synapse.nestml"],
                              target_path="/tmp/nestml-jit",
                              logging_level="INFO",
                              module_name="nestml_jit_module",
                              suffix="_nestml",
-                             codegen_opts={"neuron_parent_class": "StructuralPlasticityNode",
-                                           "neuron_parent_class_include": "structural_plasticity_node.h",
-                                           "neuron_synapse_pairs": [{"neuron": "iaf_psc_exp",
-                                                                     "synapse": "stdp",
-                                                                     "post_ports": ["post_spikes"]}]})
+                             codegen_opts=jit_codegen_opts)
+
+        if nest_version.startswith("v2"):
+            non_jit_codegen_opts = {"neuron_parent_class": "Archiving_Node",
+                                    "neuron_parent_class_include": "archiving_node.h"}
+        else:
+            non_jit_codegen_opts = {"neuron_parent_class": "ArchivingNode",
```

Comment:
Shouldn't this be `StructuralPlasticityNode`?

Reply:
It's the "non-jit" version, which tests things using the "old" ``ArchivingNode``.

Reply:
Is there a need for the if condition then? The if and else have exactly the same statements here.

Reply:
Note the presence/lack of underscoren in the class name ``Archiving_Node``.

Reply:
Got it, my bad!